### PR TITLE
refactor: rename asha to sha

### PR DESCRIPTION
**Pull request by liamcli**
## Description
Rename AsyncHalvingSearch to SyncHalvingSearch since that is what we are actually doing.  This PR will be followed up with PRs to add AsyncHalvingSearch as a beta feature alongside our current adaptive.  

## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: master/pkg/searcher/sha_test.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 )
 
 func TestASHASearcher(t *testing.T) {
```
Comment (dzhu) [inline master/pkg/searcher/sha_test.go:9]: blocking:
```suggestion
func TestSHASearcher(t *testing.T) {
```

File: master/pkg/searcher/sha_test.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ func TestASHASearcher(t *testing.T) {
 		toKinds("12S 1V 38S 1V"),
 		toKinds("12S 1V 38S 1V 150S 1V 600S 1V"),
 	}
-	checkSimulation(t, newAsyncHalvingSearch(actual), nil, ConstantValidation, expected)
+	checkSimulation(t, newSyncHalvingSearch(actual), nil, ConstantValidation, expected)
 }
 
 func TestASHASearchMethod(t *testing.T) {
```
Comment (dzhu) [inline master/pkg/searcher/sha_test.go:28]: blocking:
```suggestion
func TestSHASearchMethod(t *testing.T) {
```

File: master/pkg/model/searcher_config.go [hunk 3]
```diff
@@ -23,7 +23,7 @@ type SearcherConfig struct {
 	SingleConfig         *SingleConfig         `union:"name,single" json:"-"`
 	RandomConfig         *RandomConfig         `union:"name,random" json:"-"`
 	GridConfig           *GridConfig           `union:"name,grid" json:"-"`
-	AsyncHalvingConfig   *AsyncHalvingConfig   `union:"name,async_halving" json:"-"`
+	SyncHalvingConfig    *SyncHalvingConfig    `union:"name,sync_halving" json:"-"`
```
Comment (dzhu) [inline master/pkg/model/searcher_config.go:26]: thought: I guess I should point out that this is technically a user-facing, non-backward-compatible change, but the name is not actually documented, so it should probably be fine.

File: master/pkg/searcher/sha.go [hunk 4]
```diff
@@ -25,7 +25,7 @@ type asyncHalvingSearch struct {
 
 const ashaExitedMetricValue = math.MaxFloat64
```
Comment (dzhu) [inline master/pkg/searcher/sha.go:26]: blocking:
```suggestion
const shaExitedMetricValue = math.MaxFloat64
```

Comment (dzhu) [inline master/pkg/searcher/sha.go:26]: blocking:
```suggestion
const shaExitedMetricValue = math.MaxFloat64
```

File: master/pkg/model/searcher_config.go (hunk 3 above)
Comment (liamcli) [reply master/pkg/model/searcher_config.go:26]: note: i didn't find any experiment configs using this searcher in our examples.

**Pull request merged**